### PR TITLE
Optimize the performance of model.HashCode

### PR DESCRIPTION
**Pull request by wulianglongrd**
**Please provide a description of this PR:**
issue #41603 

xxhash.Sum64 instead of md5.Sum for better performance.
benchmark：
```code
name                      old time/op    new time/op    delta
HashCode/small_string-4      381ns ±11%     115ns ±22%   -69.80%  (p=0.000 n=9+10)
HashCode/middle_string-4     488ns ±24%     204ns ±17%   -58.17%  (p=0.000 n=9+10)
HashCode/long_string-4       659ns ±13%     279ns ±11%   -57.71%  (p=0.000 n=10+8)

name                      old alloc/op   new alloc/op   delta
HashCode/small_string-4      16.0B ± 0%     16.0B ± 0%      ~     (all equal)
HashCode/middle_string-4     16.0B ± 0%    128.0B ± 0%  +700.00%  (p=0.000 n=10+10)
HashCode/long_string-4        160B ± 0%      288B ± 0%   +80.00%  (p=0.000 n=10+10)

name                      old allocs/op  new allocs/op  delta
HashCode/small_string-4       1.00 ± 0%      1.00 ± 0%      ~     (all equal)
HashCode/middle_string-4      1.00 ± 0%      2.00 ± 0%  +100.00%  (p=0.000 n=10+10)
HashCode/long_string-4        3.00 ± 0%      2.00 ± 0%   -33.33%  (p=0.000 n=10+10)
```

### CODE REVIEW

File: pilot/pkg/model/config.go [hunk 1]
```diff
@@ -58,12 +57,15 @@ type ConfigKey struct {
 }
 
 func (key ConfigKey) HashCode() ConfigHash {
-	hash := md5.New()
-	hash.Write([]byte{byte(key.Kind)})
-	hash.Write([]byte(key.Name))
-	hash.Write([]byte(key.Namespace))
-	sum := hash.Sum(nil)
-	return ConfigHash(binary.BigEndian.Uint64(sum))
+	var builder strings.Builder
```
Comment (howardjohn) [inline pilot/pkg/model/config.go:60]: why use a String if we convert to a Byte latter? isn't it cheaper to use a `[]byte` directly?

**Comment by hzxuzhonghu** [comment]
This increases memory allocations from benchmark

File: pilot/pkg/model/config.go (hunk 1 above)
Comment (sschepens) [reply pilot/pkg/model/config.go:60]: Why not use the same approach we're already using? You can use `xxhash.New()` and the rest should be the same.

Comment (sschepens) [reply pilot/pkg/model/config.go:60]: You can also do `digest.WriteString()` you avoid converting to byte arrays when writing to the digest

Comment (wulianglongrd) [reply pilot/pkg/model/config.go]: Thanks for reminding me. That's fine, and I've committed the changes.

new benchmark
```code
name                      old time/op    new time/op    delta
HashCode/small_string-4      197ns ±13%      46ns ± 8%   -76.79%  (p=0.000 n=9+10)
HashCode/middle_string-4     298ns ±14%      66ns ±15%   -77.72%  (p=0.000 n=10+10)
HashCode/long_string-4       538ns ±20%     229ns ±39%   -57.49%  (p=0.000 n=10+10)

name                      old alloc/op   new alloc/op   delta
HashCode/small_string-4      16.0B ± 0%      0.0B       -100.00%  (p=0.000 n=10+10)
HashCode/middle_string-4     16.0B ± 0%      0.0B       -100.00%  (p=0.000 n=10+10)
HashCode/long_string-4        160B ± 0%      144B ± 0%   -10.00%  (p=0.000 n=10+10)

name                      old allocs/op  new allocs/op  delta
HashCode/small_string-4       1.00 ± 0%      0.00       -100.00%  (p=0.000 n=10+10)
HashCode/middle_string-4      1.00 ± 0%      0.00       -100.00%  (p=0.000 n=10+10)
HashCode/long_string-4        3.00 ± 0%      2.00 ± 0%   -33.33%  (p=0.000 n=10+10)
```

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**